Projects/Contracts/Sources/Viewables/ContractTableFooter.swift

```diff
@@ -20,17 +21,13 @@ extension ContractTableFooter: Viewable {
         bag += form.append(UpsellingFooter())
 
         bag +=
-            client.watch(
-                query: GraphQL.ContractsQuery(
-                    locale: Localization.Locale.currentLocale.asGraphQLLocale()
-                ),
-                cachePolicy: .fetchIgnoringCacheData
-            )
-            .compactMap { $0.contracts }.delay(by: 0.5)
-            .onValueDisposePrevious { contracts in let innerBag = DisposeBag()
-                let terminatedContractsCount = contracts.filter { $0.status.asTerminatedStatus != nil }
+            store.stateSignal.atOnce()
+            .onValueDisposePrevious { state in
+                let innerBag = DisposeBag()
+
+                let terminatedContractsCount = state.contracts.filter { $0.currentAgreement.status == .terminated }
                     .count
-                let activeContractsCount = contracts.filter { $0.status.asTerminatedStatus == nil }
+                let activeContractsCount = state.contracts.filter { $0.currentAgreement.status == .active }
```

Comment:
Would it be bad to add convenience computed getters directly on the state for common things? like `func numberOfContracts(filter: StatusFilter)`.

Reply:
Might be nice!